sdk/core/azure_core_test/src/lib.rs

```diff
@@ -52,24 +70,147 @@ impl TestContext {
             .expect("not recording or playback started")
     }
 
+    /// Gets the repository root.
+    pub fn repo_dir(&self) -> &'static Path {
+        self.repo_dir
+    }
+
+    /// Gets the service directory containing the current test.
+    ///
+    /// This is the directory under `sdk/` within the repository e.g., "core" in `sdk/core`.
+    pub fn service_directory(&self) -> &'static str {
+        self.service_directory
+    }
+
     /// Gets the test data directory under [`Self::crate_dir`].
+    ///
+    /// The path is relative to the repository root e.g., `sdk/core/azure_core/tests/data`.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the [`TestContext::crate_dir()`] is not rooted within a Git repository.
     pub fn test_data_dir(&self) -> PathBuf {
-        self.crate_dir.join("tests/data")
+        self.crate_dir
+            .join("tests/data")
+            .strip_prefix(self.repo_dir)
+            .expect("not rooted within repo")
```

Comment:
Is this error message enough to diagnose what is misconfigured?

Reply:
In context I think so. The major thing is to use unique strings people can find.